### PR TITLE
Refactor/monitor rpcs

### PR DESCRIPTION
**Pull request by adonagy**
TODO: switch base to feature/python-feature-compatibility after tests

### CODE REVIEW

File: rpc/src/rpc_actor.rs [hunk 1]
```diff
@@ -19,7 +19,8 @@ use tezos_messages::p2p::encoding::version::NetworkVersion;
 use tezos_wrapper::TezosApiConnectionPool;
 
 use crate::encoding::base_types::TimeStamp;
-use crate::helpers::current_time_timestamp;
+// TODO: move this fn
```
Comment (bkontur) [inline rpc/src/rpc_actor.rs:22]: @adonagy 
move todo `rpc/src/lib.sk`

File: rpc/src/services/stream_services.rs [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
+                to_yield_string = to_yield_string.replace("\\", "");
+                to_yield_string.push('\n');
+                Poll::Ready(Some(Ok(to_yield_string)))
+            }
+        } else {
+            let mut streamed_operations = HashSet::<String>::new();
+            let to_yield: Vec<MonitoredOperation> = requested_ops.into_iter()
+                .map(|(k, v)| {
+                    streamed_operations.insert(k);
+                    let mut monitor_op: MonitoredOperation = match serde_json::from_value(v.clone()) {
```
Comment (bkontur) [inline rpc/src/services/stream_services.rs:157]: @adonagy 
remove `v.clone()` pls.

File: rpc/src/rpc_actor.rs (hunk 1 above)
Comment (adonagy) [reply rpc/src/rpc_actor.rs]: removed

File: rpc/src/services/stream_services.rs (hunk 2 above)
Comment (adonagy) [reply rpc/src/services/stream_services.rs]: removed

File: rpc/src/services/stream_services.rs [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+                    // No current head found, storage not ready yet 
+                    Poll::Ready(None)
+                }
+            }
+        }
+    }
+}
+
+/// Get information about current head monitor header as a stream of Json strings
+pub(crate) fn get_current_head_monitor_header(chain_id: &ChainId, env: &RpcServiceEnvironment, protocol: Option<String>) -> Option<HeadMonitorStream> {
```
Comment (bkontur) [inline rpc/src/services/stream_services.rs:344]: @adonagy 
one more thing, change pls. this:
`protocol: Option<String>` -> `protocol: Option<ProtocolHash>`

**Pull request merged**